designer/server/src/routes/file/file.js

```diff
@@ -33,7 +35,19 @@ export default [
           const validation = yar.flash(
             sessionNames.validationFailure.fileDownload
           )[0]
-          return h.view('file/download-page', file.fileViewModel(validation))
+
+          const { credentials } = auth
+          const userId = hasUser(credentials) ? credentials.user.id : ''
+
+          const email =
+            (await server.methods.state.get(
+              userId,
+              sessionNames.fileDownloadPassword
+            )) ?? ''
+          return h.view(
+            'file/download-page',
+            file.fileViewModel(email, validation)
+          )
```

Comment:
I like it in a way but I am not sure if its "clean" to add normal data to validation object. I am going to make the change anyway, let me know what you think!

Reply:
That's fair. Alternatively an entire object would be consistent with other view models

E.g. Like passing `metadata` into `titleViewModel(metadata, validation)` we could pass in `file`

```patch
-  file.fileViewModel(validation)
+  file.fileViewModel(file, session)
```

Giving us access to `file.password` and `file.XXX` in future

Either works for me

Reply:
I think I will leave it as it is now but I agree adding file if we need more data is good

Reply:
I am having other issues with below when validation is null. I am reverting to what it was which seems better instead of making more changes

`if (email && !validation.formValues.email) {
            validation.formValues.email = email
          }`